libyul/SideEffects.h

```diff
@@ -0,0 +1,78 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <set>
+
+namespace yul
+{
+
+/**
+ * Side effects of code.
+ *
+ * The default-constructed value applies to the "empty code".
+ */
+struct SideEffects
+{
+	/// If true, expressions in this code can be freely moved and copied without altering the
+	/// semantics.
+	/// At statement level, it means that functions containing this code can be
+	/// called multiple times, their calls ran be rearranged and calls can also be
+	/// deleted without changing the semantics.
+	/// This means it cannot depend on storage or memory, cannot have any side-effects,
+	/// but it can depend on state that is constant across an EVM-call.
+	bool movable = true;
```

Comment:
Why do they now start with the best case side effects?

Reply:
The default-constructed side-effects correspond to "empty code".